modin/core/dataframe/base/dataframe/dataframe.py

```diff
@@ -0,0 +1,502 @@
+        col_labels: Optional[List[Hashable]] = None,
+        col_positions: Optional[List[int]] = None,
+    ) -> "ModinDataframe":
+        """
+        Mask rows and columns in the dataframe.
+
+        Allow users to perform selection and projection on the row and column number (positional notation),
+        in addition to the row and column labels (named notation).
+
+        Parameters
+        ----------
+        row_labels : list of hashable, optional
+            The row labels to extract.
+        row_positions : list of int, optional
+            The row indices to extract.
+        col_labels : list of hashable, optional
+            The column labels to extract.
+        col_positions : list of int, optional
+            The column indices to extract.
+
+        Returns
+        -------
+        ModinDataframe
+             A new ModinDataframe from the mask provided.
+        """
+        pass
+
+    @abstractmethod
+    def filter_by_types(self, types: List[Hashable]) -> "ModinDataframe":
+        """
+        Allow the user to specify a type or set of types by which to filter the columns.
+
+        Parameters
+        ----------
+        types : list of hashables
+            The types to filter columns by.
+
+        Returns
+        -------
+        ModinDataframe
+             A new ModinDataframe with only the columns whose dtypes appear in types.
+        """
+        pass
+
+    @abstractmethod
+    def map(
+        self,
+        function: Callable,
+        axis: Optional[int] = None,
+        dtypes: Optional[str] = None,
+    ) -> "ModinDataframe":
+        """
+        Apply a user-defined function row-wise if axis=0, column-wise if axis=1, and cell-wise if axis is None.
+
+        Parameters
+        ----------
+        function : callable
+            The function to map across the dataframe.
+        axis : int, optional
+            The axis to map over.
+        dtypes : str, optional
+            The data types for the result. This is an optimization
+            because there are functions that always result in a particular data
+            type, and this allows us to avoid (re)computing it.
+
+        Returns
+        -------
+        ModinDataframe
+             A new ModinDataframe with the map applied.
+
+        Notes
+        -----
+        This does not change the shape of the dataframe.
+        """
+        pass
+
+    @abstractmethod
+    def filter(self, axis: int, condition: Callable) -> "ModinDataframe":
+        """
+        Filter data based on the function provided along the specified axis.
+
+        Parameters
+        ----------
+        axis : int
+            The axis to filter over.
+        condition : callable
+            The function to use for the filter. This function should filter the
+            data itself. It accepts either a row or column (depending on the axis argument) and returns True to keep the row/col, otherwise False.
```

Comment:
nit: add something like "... otherwise False to filter out the row/col."

makes it clear that True = keep, and False = filter out.